libFMS_mod/libFMS.F90

```diff
@@ -0,0 +1,471 @@
+
+  !> fms2_io
+  use fms2_io_mod, only: unlimited, FmsNetcdfFile_t, FmsNetcdfDomainFile_t, &
+                         FmsNetcdfUnstructuredDomainFile_t, open_file, open_virtual_file, &
+                         close_file, register_axis, register_field, register_restart_field, &
+                         write_data, read_data, write_restart, write_new_restart, &
+                         read_restart, read_new_restart, global_att_exists, &
+                         variable_att_exists, register_global_attribute, &
+                         register_variable_attribute, get_global_attribute, &
+                         get_variable_attribute, get_num_dimensions, &
+                         get_dimension_names, dimension_exists, is_dimension_unlimited, &
+                         get_dimension_size, get_num_variables, get_variable_names, &
+                         variable_exists, get_variable_num_dimensions, &
+                         get_variable_dimension_names, get_variable_size, &
+                         get_compute_domain_dimension_indices, &
+                         get_global_io_domain_indices, Valid_t, get_valid, is_valid, &
+                         get_unlimited_dimension_name, get_variable_unlimited_dimension_index, &
+                         file_exists, compressed_start_and_count, get_variable_sense, &
+                         get_variable_missing, get_variable_units, get_time_calendar, &
+                         open_check, is_registered_to_restart, check_if_open, &
+                         set_fileobj_time_name, is_dimension_registered, &
+                         fms2_io_init, get_mosaic_tile_grid
+  ! used via fms2_io
+  ! fms_io_utils_mod, fms_netcdf_domain_io_mod, netcdf_io_mod, 
+  ! fms_netcdf_unstructured_domain_io_mod, blackboxio
+
+  !> fms
+  !! routines that don't conflict with fms2_io
+  use fms_mod, only: fms_init, fms_end, error_mesg, fms_error_handler, check_nml_error, &
+                     monotonic_array, string_array_index
+
+  !> horiz_interp
+  use horiz_interp_mod, only: horiz_interp, horiz_interp_new, horiz_interp_del, &
+                              horiz_interp_init, horiz_interp_end
+  use horiz_interp_type_mod, only: horiz_interp_type, assignment(=), CONSERVE, &
+                              BILINEAR, SPHERICA, BICUBIC, stats
+  !! used via horiz_interp
+  ! horiz_interp_bicubic_mod, horiz_interp_bilinear_mod
+  ! horiz_interp_conserve_mod, horiz_interp_spherical_mod
+
+  !> interpolator
+  use interpolator_mod, only: interpolator_init, interpolator, interpolate_type_eq, &
+                              obtain_interpolator_time_slices, unset_interpolator_time_flag, &
+                              interpolator_end, init_clim_diag, query_interpolator, &
+                              interpolate_type, CONSTANT, &
+                              INTERP_WEIGHTED_P, INTERP_LINEAR_P, INTERP_LOG_P, &
+                              ZERO_INTERP=>ZERO, & !! conflicts with mpp_parameter's ZERO 
```

Comment:
The fact we have a parameter named ZERO in interpolator_mod that is given a value of '2' is a real danger.  If the ZERO parameter from this module is not used in any of our actual model code, we should remove it's public specification and rename it internally to make it's use more clear.

Reply:
It looks an array of the ZERO or CONSTANT flags is used for calling interpolator_init, so I think it may need to stay public to use that routine.